.github/workflows/build_and_test.yml

```diff
@@ -5,11 +5,46 @@ on:
     branches:
     - '**'
     - '!branch-*.*'
+  schedule:
+    # master
+    - cron: '0 0 * * *'
+    # branch-3.2
+    - cron: '0 1 * * *'
 
 jobs:
+  configure-jobs:
+    name: Configure jobs
+    runs-on: ubuntu-20.04
+    outputs:
+      should_run: ${{ steps.set-outputs.outputs.should_run }}
+      branch: ${{ steps.set-outputs.outputs.branch }}
+      envs: ${{ steps.set-outputs.outputs.envs }}
+    env:
+      SCHEDULE: ${{ github.event.schedule }}
+    steps:
+    - name: Configure branch and additional environment variables
+      id: set-outputs
+      run: |
+        if [ "$SCHEDULE" = "0 0 * * *" ]; then
+          echo '::set-output name=branch::master'
+          echo '::set-output name=envs::{"SCALA_PROFILE": "scala2.13"}'
+          echo '::set-output name=should_run::${{ github.repository == 'apache/spark' }}'
+        elif [ "$SCHEDULE" = "0 1 * * *" ]; then
+          echo '::set-output name=branch::branch-3.2"
+          echo '::set-output name=envs::{"SCALA_PROFILE": "scala2.13"}'
+          echo '::set-output name=should_run::${{ github.repository == 'apache/spark' }}'
+        else
+          echo '::set-output name=branch::master' # Default branch to run on. CHANGE here when a branch is cut out.
+          echo '::set-output name=envs::{}'
+          echo '::set-output name=should_run::true'
+        fi
+
   # Build: build Spark and run the tests for specified modules.
   build:
     name: "Build modules: ${{ matrix.modules }} ${{ matrix.comment }} (JDK ${{ matrix.java }}, ${{ matrix.hadoop }}, ${{ matrix.hive }})"
```

Comment:
Could you expose `branch` here? The existing `build_and_test_scala2.13_daily.yml` exposes the branch name. Without branch changes, it was confusing at that time. For this job, I'm wondering how it is distinguished.

Reply:
Oh, actually I added `caption` for that purpose. It will be shown in the job name like:

<img width="749" alt="Screen Shot 2021-07-19 at 2 30 01 PM" src="https://user-images.githubusercontent.com/6477701/126107646-040c9252-3371-4ff2-94ea-95f59dac34b0.png">

(ref: https://github.community/t/github-actions-dynamic-name-of-the-workflow-with-workflow-dispatch/150327)

Reply:
and probably we could use this to filter out:
<img width="405" alt="Screen Shot 2021-07-19 at 2 32 23 PM" src="https://user-images.githubusercontent.com/6477701/126107839-b0ab5def-1ca5-4715-be87-f968886a4744.png">